Add `onLoad` gottcha note to `next/script` docs (#33097)
vercel/next.js: docs/api-reference/next/script.md

A method that returns additional JavaScript that should be executed after the script has finished loading.

> > **Note: onLoad will not fire if the `strategy` prop is set to `beforeInteractive`.**

Reviewer: ```suggestion
> **Note: `onLoad` can't be used with the `beforeInteractive` loading strategy.**
```
